File: Programming/Python/LeibnizFormula/leibniz_formula.py
def leibniz(k: int) -> float:
	sum = 0.0
	for i in range(k):
		sum += leibniz_step(i)

	return sum*4


def leibniz_step(k: int) -> float:
	return (-1)**k / (2*k + 1)


if __name__ == '__main__':
	print(leibniz(10000))

File: Programming/Java/LeibnizFormula/LeibnizFormula.java
public class LeibnizFormula {

    public static void main(String[] args) {
        System.out.println(leibniz(100000));
    }

    public static double leibniz(int k) {
        double sum = 0;
        for(int i=0; i<k; i++) {
            sum += leibnizStep(i);
        }

        return sum*4;
    }

    private static double leibnizStep(int k) {
        return Math.pow(-1, k) / (2*k +1);
    }
}
